### PR TITLE
Refactor the firewall denied error condition and add a test

### PR DESCRIPTION
**Pull request by lincoln-replit**
Why
===

Discussed in https://github.com/replit/crosis/pull/128

In order to present a more friendly interface to the user when this happens, so that firewalledreplit.com is not perceived as buggy, let's give ourselves an explicit function to call

Also, add a test so we know it works

What changed
============

- Instead of throwing an error, call a function (still throws an error by default)
- Small test refactor so we can put the new test in its own file

Test plan
=========

Test should be sufficient

### CODE REVIEW

**Comment by lincoln-replit** [comment]
Fixed goval, this works now!

**Review by masad-frost — APPROVED** [review]
Best javascript slinger in the south. Thanks for following up.

File: debug/genConnectionMetadata.d.ts [hunk 1]
```diff
@@ -1,5 +1,17 @@
-export default function genConnectionMetadata(): {
+type Repl = {
+  id: string;
+  languag: string;
```
Comment (xyc) [inline debug/genConnectionMetadata.d.ts:3]: typo?
```suggestion
  language: string;
```

File: debug/genConnectionMetadata.js [hunk 2]
```diff
@@ -20,9 +20,22 @@ ${keyData.toString('base64')}
...
+  if (repl == null) {
+    repl = {
+      id: `testing-crosis-${Math.random().toString(36).split('.')[1]}`,
+      language: 'bash',
+      slug: Math.random().toString(36).slice(2),
+      user: 'crosistest',
+      bucket: 'test-replit-repls',
+    };
+  }
+
```
Comment (xyc) [inline debug/genConnectionMetadata.js:38]: nit: slightly easier way of writing
```suggestion
  let repl = options?.repl || {
    id: `testing-crosis-${Math.random().toString(36).split('.')[1]}`,
    language: 'bash',
    slug: Math.random().toString(36).slice(2),
    user: 'crosistest',
    bucket: 'test-replit-repls',
  };
```

File: src/__testutils__/done.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/* eslint-env jest */
+
+/**
+ * A simple helper to allow us to do assertions inside callbacks.
+ * If `done` is not called after an assertion fails inside a callback
+ * the test will continue until it times out. This function makes sure
+ * done is called and we fail fast with correct stack traces.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function wrapWithDone<Args extends Array<any>, Ret>(
```
Comment (xyc) [inline src/__testutils__/done.ts:10]: nice refactor

**Pull request merged**